src/axom/slam/examples/HandleMesh.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2017-2019, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * \file HandleMesh.cpp
+ *
+ * \brief Slam example where the ElementType of a set is a handle
+ * that wraps a mesh index.
+ */
+
+#include "axom/slic.hpp"
+#include "axom/slam.hpp"
+
+#include <ostream>
+
+namespace slam = axom::slam;
+namespace slic = axom::slic;
+
+
+namespace
+{
+
+/**
+ * \class Handle
+ * \brief A handle is a wrapper around a variable.
+ *
+ * This class is a stub for some mesh data structures
+ * that use typed index handles rather than (untyped) indices
+ * to refer to mesh elements.
+ */
+template<typename T>
+struct Handle
+{
+  Handle() : mID(T()) {}
+  explicit Handle(T id) : mID(id) {}
+  Handle(const Handle& h) : mID(h.mID) {}
+  bool operator==(const Handle& h) { return mID == h.mID; }
+
+  static Handle make_handle(T id) { return Handle(id); }
+
+  void print(std::ostream& os) const
+  {
+    os << "Handle(" << mID <<")";
+  }
+
+  // Simple function to call on a Handle
+  T twiceIndex() const { return 2 * mID; }
+
+  T mID;
+};
+
+template<typename T>
+std::ostream& operator<<(std::ostream& os, const Handle<T>& h)
+{
+  h.print(os);
+  return os;
+}
+
+} // end anonymous namespace
+
+
+int main(int, char**)
+{
+  slic::UnitTestLogger logger;
+
+  using PosType = slam::PositionType;
+  using HandleType = Handle<PosType>;
+
+  using HandleSet = slam::VectorIndirectionSet<PosType, HandleType>;
+
+  const int sz = 5;
+  HandleSet::IndirectionBufferType vecHandle(sz);
+
+  // Create a set of handles
+  HandleSet hSet = HandleSet::SetBuilder()
+                   .size( sz )
+                   .data( &vecHandle );
+
+  // Add handles with (somewhat) arbitrary IDs to the set
+  for(auto i : hSet.positions() )
```

Comment:
This is also new -- I added a `positions()` function to `slam::OrderedSet`, which makes it easy to iterate over the indices (from `0` to `set.size()`). We can then access the set element using the subscript operator, e.g. `hSet[i]`.